.tools/validate.go

```diff
@@ -0,0 +1,269 @@
+				fmt.Println("PASS")
+				if *flVerbose {
+					for _, r := range vr {
+						if !r.Pass {
+							fmt.Printf("  - %s\n", r.Msg)
+						}
+					}
+				}
+			} else {
+				fmt.Println("FAIL")
+				// default, only print out failed validations
+				for _, r := range vr {
+					if !r.Pass {
+						fmt.Printf("  - %s\n", r.Msg)
+					}
+				}
+			}
+		}
+	}
+	_, fail := results.PassFail()
+	if fail > 0 {
+		fmt.Printf("%d issues to fix\n", fail)
+		os.Exit(1)
+	}
+}
+
+// ValidateRule will operate over a provided CommitEntry, and return a result.
+// An error should abort further tests.
+type ValidateRule func(CommitEntry) (ValidateResult, error)
+
+// ValidateCommit processes the given rules on the provided commit, and returns the result set.
+// An error should abort further tests.
+func ValidateCommit(c CommitEntry, rules []ValidateRule) (ValidateResults, error) {
+	results := ValidateResults{}
+	for _, r := range rules {
+		vr, err := r(c)
+		if err != nil {
+			return results, err
+		}
+		results = append(results, vr)
+	}
+	return results, nil
+}
+
+// ValidateResult is the result for a single validation of a commit.
+type ValidateResult struct {
+	CommitEntry CommitEntry
+	Pass        bool
+	Msg         string
+}
+
+// ValidateResults is a set of results. This is type makes it easy for the following function.
+type ValidateResults []ValidateResult
+
+// PassFail gives a quick over/under of passes and failures of the results in this set
+func (vr ValidateResults) PassFail() (pass int, fail int) {
+	for _, res := range vr {
+		if res.Pass {
+			pass++
+		} else {
+			fail++
+		}
+	}
+	return pass, fail
+}
+
+// CommitEntry represents a single commit's information from `git`
+type CommitEntry struct {
+	Commit               string
+	AbbreviatedCommit    string `json:"abbreviated_commit"`
+	Tree                 string
+	AbbreviatedTree      string `json:"abbreviated_tree"`
+	Parent               string
+	AbbreviatedParent    string `json:"abbreviated_parent"`
+	Refs                 string
+	Encoding             string
+	Subject              string
+	SanitizedSubjectLine string `json:"sanitized_subject_line"`
+	Body                 string
+	CommitNotes          string `json:"commit_notes"`
+	VerificationFlag     string `json:"verification_flag"`
+	ShortMsg             string
+	Signer               string
+	SignerKey            string       `json:"signer_key"`
+	Author               PersonAction `json:"author,omitempty"`
+	Commiter             PersonAction `json:"commiter,omitempty"`
+}
+
+// PersonAction is a time and identity of an action on a git commit
+type PersonAction struct {
+	Name  string `json:"name"`
+	Email string `json:"email"`
+	Date  string `json:"date"` // this could maybe be an actual time.Time
+}
+
+var (
+	prettyLogSubject       = `--pretty=format:%s`
+	prettyLogBody          = `--pretty=format:%b`
+	prettyLogCommit        = `--pretty=format:%H`
+	prettyLogAuthorName    = `--pretty=format:%aN`
```

Comment:
I'm not sure how old Travis' Git is (and I haven't looked up `--format` in Git's release notes), but at least with v2.1 you can use `--format=…` instead of `--pretty=format:…`.

I also think it makes sense to drop the leading `--pretty=format:` (or whatever) here and just list `%s`, `%b`, etc.  We can inject the option prefix once in the consuming code.